Building with 3.0.2 fails on Windows if there's a `build-figma-plugin.ui.cjs` file in the repo

I think this is #77 all over again.  The `build-figma-plugin.ui.cjs` file is just this:

```js
module.exports = (buildOptions) => ({
	...buildOptions,
	keepNames: true,
});
```

When it's in the main directory, `npm run build` fails with this error:

```sh
info Typechecking...
success Typechecked in 2.048s
info Building...
error esbuild error
    Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
npm ERR! Lifecycle script `build` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: fwidgets-test-plugin@0.0.1 
npm ERR!   at location: C:\...\test 
```

When the config file is removed, the build works, but the plugin doesn't, because it needs to pass `keepNames: true` to esbuild.  (As an aside, it would be nice if this sort of config, as well as the manifest, could be passed on the command line, which would then avoid this problem.)

--- a/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
+++ b/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
@@ -18,7 +18,7 @@ import { importFresh } from '../import-fresh.js'
 import { esbuildCssModulesPlugin } from './esbuild-css-modules-plugin.js'
 import { esbuildPreactCompatPlugin } from './esbuild-preact-compat-plugin.js'
 
-const isWindows = platform() === 'win32';
+const isWindows = platform() === 'win32'
 
 interface EntryFile extends ConfigFile {
   commandId: string
@@ -57,7 +57,7 @@ async function overrideEsbuildConfigAsync(
     return buildOptions
   }
   if (isWindows) {
-    filePaths = filePaths.map(p => pathToFileURL(p).href);
+    filePaths = filePaths.map(p => pathToFileURL(p).href)
   }
   const overrideEsbuildConfig:
     | OverrideEsbuildConfig

--- a/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
+++ b/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
@@ -1,4 +1,5 @@
 import { resolve } from 'node:path'
+import { platform } from 'node:os'
 
 import {
   Config,
@@ -48,9 +49,12 @@ async function overrideEsbuildConfigAsync(
   buildOptions: BuildOptions,
   configGlobPattern: string
 ): Promise<BuildOptions> {
-  const filePaths = await globby(configGlobPattern, { absolute: true })
+  let filePaths = await globby(configGlobPattern, { absolute: true })
   if (filePaths.length === 0) {
     return buildOptions
+  }
+  if (platform() === 'win32') {
+    filePaths = `file://${filePaths}`;
   }
   const overrideEsbuildConfig:
     | OverrideEsbuildConfig

--- a/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
+++ b/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
@@ -54,7 +54,7 @@ async function overrideEsbuildConfigAsync(
     return buildOptions
   }
   if (platform() === 'win32') {
-    filePaths = `file://${filePaths}`;
+    filePaths = filePaths.map(p => `file:///${p}`);
   }
   const overrideEsbuildConfig:
     | OverrideEsbuildConfig

--- a/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
+++ b/packages/build/src/utilities/build-bundles-async/build-bundles-async.ts
@@ -1,5 +1,6 @@
 import { resolve } from 'node:path'
 import { platform } from 'node:os'
+import { pathToFileURL } from 'node:url'
 
 import {
   Config,
@@ -16,6 +17,8 @@ import indentString from 'indent-string'
 import { importFresh } from '../import-fresh.js'
 import { esbuildCssModulesPlugin } from './esbuild-css-modules-plugin.js'
 import { esbuildPreactCompatPlugin } from './esbuild-preact-compat-plugin.js'
+
+const isWindows = platform() === 'win32';
 
 interface EntryFile extends ConfigFile {
   commandId: string
@@ -53,8 +56,8 @@ async function overrideEsbuildConfigAsync(
   if (filePaths.length === 0) {
     return buildOptions
   }
-  if (platform() === 'win32') {
-    filePaths = filePaths.map(p => `file:///${p}`);
+  if (isWindows) {
+    filePaths = filePaths.map(p => pathToFileURL(p).href);
   }
   const overrideEsbuildConfig:
     | OverrideEsbuildConfig

--- a/packages/build/src/utilities/build-manifest-async.ts
+++ b/packages/build/src/utilities/build-manifest-async.ts
@@ -1,6 +1,8 @@
 /* eslint-disable sort-keys-fix/sort-keys-fix */
 
 import { resolve } from 'node:path'
+import { platform } from 'node:os'
+import { pathToFileURL } from 'node:url'
 
 import {
   Config,
@@ -21,6 +23,8 @@ import {
 import { globby } from 'globby'
 
 import { importFresh } from './import-fresh.js'
+
+const isWindows = platform() === 'win32'
 
 export async function buildManifestAsync(options: {
   config: Config
@@ -209,11 +213,14 @@ function createManifestNetworkAccess(
 async function overrideManifestAsync(
   manifest: Manifest
 ): Promise<Record<string, any>> {
-  const filePaths = await globby(constants.build.manifestConfigGlobPattern, {
+  let filePaths = await globby(constants.build.manifestConfigGlobPattern, {
     absolute: true
   })
   if (filePaths.length === 0) {
     return manifest
+  }
+  if (isWindows) {
+    filePaths = filePaths.map(p => pathToFileURL(p).href)
   }
   const { default: overrideManifest } = await importFresh(filePaths[0])
   return overrideManifest(manifest)

